### PR TITLE
feat(common-nest-services): Initial boilerplate

### PR DESCRIPTION
**Pull request by atharmon**
Generated boilerplate for:
- `libs/common/nest/services`

### CODE REVIEW

**Comment by Proantagonist** [comment]
What's the rationale between putting this in a standalone lib `common-nest-services` as opposed to exporting the service from the `@tamu-gisc/mailroom` scope?

File: libs/common/nest/services/src/lib/mailer/mailer.service.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import { EnvironmentService } from '@tamu-gisc/common/nest/environment';
+import { IMailroomEmailOutbound } from '@tamu-gisc/mailroom/common';
+
+@Injectable()
+export class MailerService {
+  constructor(private readonly env: EnvironmentService) {}
+
+  public sendMail(mailConfig: IMailroomEmailOutbound) {
+    // TODO: Need the environment service to get the address of the mailroom-nest instance from environment file
+    const mailroomUrl = this.env.value('mailroomUrl');
```
Comment (Proantagonist) [inline libs/common/nest/services/src/lib/mailer/mailer.service.ts:14]: Is environment service expected to be loaded on the host application? I noticed there's no environment registered on the mailer module.

File: libs/common/nest/services/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# common-nest-services
```
Comment (Proantagonist) [inline libs/common/nest/services/README.md:1]: Is documentation going to be added in another PR?

**Review by Proantagonist — COMMENTED** [review]
Only thing I would suggest is keep in mind the general scope of this library. 

What *all* is going to fall under the `common/nest/services` umbrella? It's a very broad scope.

 Would it not be better to have a more specific library `common/nest/utils/mailer`, for example?

File: libs/common/nest/services/src/lib/mailer/mailer.service.ts (hunk 1 above)
Comment (atharmon) [reply libs/common/nest/services/src/lib/mailer/mailer.service.ts:14]: At the moment yes. I would like to integrate the EnvironmentModule in a coming PR so all you have to do is provide the environment variable within the app.module and not worry about importing EnvironmentModule yourself.

**Review by Proantagonist — APPROVED** [review]
